tools/zarr-cache-stats.py

```diff
@@ -0,0 +1,299 @@
+                        key = MissKey(span_id=span_id, manifest_path=manifest_path)
+                        if (start := misses_in_progress.pop(key, None)) is not None:
+                            end = MissMoment(
+                                timestamp=timestamp,
+                                approx_cache_len=fields["approx_cache_len"],
+                            )
+                            events.misses.append(
+                                Miss(
+                                    manifest_path=manifest_path,
+                                    start=start,
+                                    end=end,
+                                    parallel=len(misses_in_progress),
+                                )
+                            )
+                        last_accesses[manifest_path] = timestamp
+                    case "hit":
+                        last_access = last_accesses.get(manifest_path)
+                        if last_access is not None:
+                            recency_index = sum(
+                                1 for ts in last_accesses.values() if ts > last_access
+                            )
+                        else:
+                            recency_index = len(last_accesses)
+                        events.hits.append(
+                            Hit(
+                                manifest_path=manifest_path,
+                                timestamp=timestamp,
+                                last_access=last_access,
+                                approx_cache_len=fields["approx_cache_len"],
+                                recency_index=recency_index,
+                            )
+                        )
+                        last_accesses[manifest_path] = timestamp
+                    case "evict":
+                        events.evictions.append(
+                            Eviction(
+                                manifest_path=manifest_path,
+                                timestamp=timestamp,
+                                cause=RemovalCause(fields["cause"]),
+                            )
+                        )
+                        last_accesses.pop(manifest_path, None)
+                    case "dump" | "dump-error":
+                        pass
+                    case other:
+                        log.warning(
+                            "Invalid 'cache_event' field value %r: %s", other, lg.line
+                        )
+        elif lg.target == "heroku[web.1]" and lg.message in (
+            "Stopping all processes with SIGTERM",
+            "Stopping process with SIGKILL",
+        ):
+            events.killed_misses.extend(
+                KilledMiss(
+                    manifest_path=k.manifest_path, start=v, killed_at=lg.timestamp
+                )
+                for k, v in misses_in_progress.items()
+            )
+            misses_in_progress = {}
+            last_accesses = {}
```

Comment:
is this the end and there should be no other hits or a new dandidav starts and we should get new separate stats or you are just collating them across all runs?

Reply:
This is the start of a new `dandidav` process, so any information specific to the previous process needs to be cleared.